cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -1124,8 +1135,17 @@ private boolean isPodUpToDate(StatefulSet ss, Pod pod) {
             return ssGeneration == podGeneration;
         }
 
+        private boolean isPodCaCertUpToDate(Pod pod) {
+            final int caCertGeneration =
+                    Integer.parseInt(clusterCa.caCertSecret().getMetadata().getAnnotations().get(Ca.ANNO_STRIMZI_IO_CA_CERT_GENERATION));
```

Comment:
You're using `clusterCa.caCertSecret().getMetadata().getAnnotations().get(Ca.ANNO_STRIMZI_IO_CA_CERT_GENERATION)` in a few places. You should factor this into a method.